### PR TITLE
Roles branch fix & tests

### PR DESCRIPTION
**Pull request by KIMdaniiell**

### CODE REVIEW

File: src/main/java/org/itmo/eventapp/main/service/RoleService.java [hunk 1]
```diff
@@ -65,10 +66,11 @@ public void deleteRole(Integer id) {
         if (role.getType().equals(RoleType.SYSTEM)) {
             if (!userService.findAllByRole(role).isEmpty())
```
Comment (anamrzv) [inline src/main/java/org/itmo/eventapp/main/service/RoleService.java:67]: по сути можно заменить findAll на exists чтобы повысить производительность 
https://www.baeldung.com/spring-data-exists-query

File: src/main/java/org/itmo/eventapp/main/service/RoleService.java [hunk 2]
```diff
@@ -79,7 +81,7 @@ public List<Role> getAll() {
 
     public Role findRoleById(Integer id) {
         return roleRepository.findById(id)
-                .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND, ExceptionConst.ROLE_NOT_FOUND_MESSAGE));
+                .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND, String.format("Роли с id %d не существует", id)));
```
Comment (anamrzv) [inline src/main/java/org/itmo/eventapp/main/service/RoleService.java:84]: почему убрали ExceptionConst? мне кажется лучше вернуть, там в строке тоже поставить %d

File: src/main/java/org/itmo/eventapp/main/service/RoleService.java [hunk 3]
```diff
@@ -90,17 +92,23 @@ public List<Role> searchByName(String name) {
         return roleRepository.findByNameContainingIgnoreCase(name);
     }
 
-    public void assignSystemRole(Integer userId, Integer roleId) {
+    public void assignSystemRole(String email, Integer userId, Integer roleId) {
         var user = userService.findById(userId);
+        var userWithEmail = userLoginInfoService.findByEmail(email);
+        if (userWithEmail.getUser().getId().equals(userId))
+            throw new ResponseStatusException(HttpStatus.FORBIDDEN, "Невозможно назначить роль себе");
```
Comment (anamrzv) [inline src/main/java/org/itmo/eventapp/main/service/RoleService.java:99]: наверное остальные сообщения тоже вынести в константу

File: src/main/java/org/itmo/eventapp/main/service/RoleService.java (hunk 2 above)
Comment (rimnvd) [reply src/main/java/org/itmo/eventapp/main/service/RoleService.java:84]: Убрала, потому что все остальное у нас было без констант и на русском, а в константах на английском, наверное, стоит все вынести в константы и договориться, на каком языке мы все-таки будем возвращать ошибки

Comment (anamrzv) [reply src/main/java/org/itmo/eventapp/main/service/RoleService.java:84]: давай перенесем обратно в константы и все сообщения там на русском сделаем

**Comment by IP-13** [comment]
|Overall Project|76.4% **`-6.34%`**|:green_apple:|
|:-|:-|:-:|
|Files changed|28.88%|:x:|
<br>

|File|Coverage||
|:-|:-|:-:|
|[UserService.java](https://github.com/itmo-events-app/event-app/blob/d54f7d4846075fed1cd0028c270dc988adac7bfe/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FUserService.java)|90.63%|:green_apple:|
|[RoleService.java](https://github.com/itmo-events-app/event-app/blob/d54f7d4846075fed1cd0028c270dc988adac7bfe/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FRoleService.java)|71.93% **`-17.98%`**|:green_apple:|
|[PrivilegeService.java](https://github.com/itmo-events-app/event-app/blob/d54f7d4846075fed1cd0028c270dc988adac7bfe/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FPrivilegeService.java)|28.85% **`-38.46%`**|:x:|
|[EventRoleService.java](https://github.com/itmo-events-app/event-app/blob/d54f7d4846075fed1cd0028c270dc988adac7bfe/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FEventRoleService.java)|23.49% **`-46.64%`**|:x:|
|[RoleController.java](https://github.com/itmo-events-app/event-app/blob/d54f7d4846075fed1cd0028c270dc988adac7bfe/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fcontroller%2FRoleController.java)|6.35% **`-38.62%`**|:x:|

**Comment by IP-13** [comment]
|Overall Project|76.4% **`-6.34%`**|:green_apple:|
|:-|:-|:-:|
|Files changed|28.88%|:x:|
<br>

|File|Coverage||
|:-|:-|:-:|
|[UserService.java](https://github.com/itmo-events-app/event-app/blob/a7cb3721c7b156b172032b0888db974f9d7ef738/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FUserService.java)|90.63%|:green_apple:|
|[RoleService.java](https://github.com/itmo-events-app/event-app/blob/a7cb3721c7b156b172032b0888db974f9d7ef738/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FRoleService.java)|71.93% **`-17.98%`**|:green_apple:|
|[PrivilegeService.java](https://github.com/itmo-events-app/event-app/blob/a7cb3721c7b156b172032b0888db974f9d7ef738/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FPrivilegeService.java)|28.85% **`-38.46%`**|:x:|
|[EventRoleService.java](https://github.com/itmo-events-app/event-app/blob/a7cb3721c7b156b172032b0888db974f9d7ef738/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FEventRoleService.java)|23.49% **`-46.64%`**|:x:|
|[RoleController.java](https://github.com/itmo-events-app/event-app/blob/a7cb3721c7b156b172032b0888db974f9d7ef738/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fcontroller%2FRoleController.java)|6.35% **`-38.62%`**|:x:|

**Comment by IP-13** [comment]
|Overall Project|76.64% **`-6.34%`**|:green_apple:|
|:-|:-|:-:|
|Files changed|28.88%|:x:|
<br>

|File|Coverage||
|:-|:-|:-:|
|[UserService.java](https://github.com/itmo-events-app/event-app/blob/46cebe98d7c245148769fb38d81de409726b4073/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FUserService.java)|90.63%|:green_apple:|
|[RoleService.java](https://github.com/itmo-events-app/event-app/blob/46cebe98d7c245148769fb38d81de409726b4073/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FRoleService.java)|71.93% **`-17.98%`**|:green_apple:|
|[PrivilegeService.java](https://github.com/itmo-events-app/event-app/blob/46cebe98d7c245148769fb38d81de409726b4073/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FPrivilegeService.java)|28.85% **`-38.46%`**|:x:|
|[EventRoleService.java](https://github.com/itmo-events-app/event-app/blob/46cebe98d7c245148769fb38d81de409726b4073/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FEventRoleService.java)|23.49% **`-46.64%`**|:x:|
|[RoleController.java](https://github.com/itmo-events-app/event-app/blob/46cebe98d7c245148769fb38d81de409726b4073/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fcontroller%2FRoleController.java)|6.35% **`-38.62%`**|:x:|

**Comment by IP-13** [comment]
|Overall Project|76.4% **`-6.34%`**|:green_apple:|
|:-|:-|:-:|
|Files changed|28.88%|:x:|
<br>

|File|Coverage||
|:-|:-|:-:|
|[UserService.java](https://github.com/itmo-events-app/event-app/blob/87fd21f1315f1283f6f3c4859d3ebac4f0edc928/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FUserService.java)|90.63%|:green_apple:|
|[RoleService.java](https://github.com/itmo-events-app/event-app/blob/87fd21f1315f1283f6f3c4859d3ebac4f0edc928/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FRoleService.java)|71.93% **`-17.98%`**|:green_apple:|
|[PrivilegeService.java](https://github.com/itmo-events-app/event-app/blob/87fd21f1315f1283f6f3c4859d3ebac4f0edc928/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FPrivilegeService.java)|28.85% **`-38.46%`**|:x:|
|[EventRoleService.java](https://github.com/itmo-events-app/event-app/blob/87fd21f1315f1283f6f3c4859d3ebac4f0edc928/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FEventRoleService.java)|23.49% **`-46.64%`**|:x:|
|[RoleController.java](https://github.com/itmo-events-app/event-app/blob/87fd21f1315f1283f6f3c4859d3ebac4f0edc928/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fcontroller%2FRoleController.java)|6.35% **`-38.62%`**|:x:|

**Comment by IP-13** [comment]
|Overall Project|77.72% **`-6.68%`**|:green_apple:|
|:-|:-|:-:|
|Files changed|28.15%|:x:|
<br>

|File|Coverage||
|:-|:-|:-:|
|[UserService.java](https://github.com/itmo-events-app/event-app/blob/d89e548c60ed8b3d5934ab8a6879b975f9c7055c/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FUserService.java)|90.63%|:green_apple:|
|[RoleService.java](https://github.com/itmo-events-app/event-app/blob/d89e548c60ed8b3d5934ab8a6879b975f9c7055c/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FRoleService.java)|71.93% **`-17.98%`**|:green_apple:|
|[PrivilegeService.java](https://github.com/itmo-events-app/event-app/blob/d89e548c60ed8b3d5934ab8a6879b975f9c7055c/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FPrivilegeService.java)|27.27% **`-65.45%`**|:x:|
|[EventRoleService.java](https://github.com/itmo-events-app/event-app/blob/d89e548c60ed8b3d5934ab8a6879b975f9c7055c/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FEventRoleService.java)|23.49% **`-46.64%`**|:x:|
|[RoleController.java](https://github.com/itmo-events-app/event-app/blob/d89e548c60ed8b3d5934ab8a6879b975f9c7055c/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fcontroller%2FRoleController.java)|6.35% **`-38.62%`**|:x:|

**Comment by IP-13** [comment]
|Overall Project|77.21% **`-5.91%`**|:green_apple:|
|:-|:-|:-:|
|Files changed|36.38%|:x:|
<br>

|File|Coverage||
|:-|:-|:-:|
|[PrivilegeService.java](https://github.com/itmo-events-app/event-app/blob/efe64da851a49447f2fa8e637e2311fb2b41fa7b/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FPrivilegeService.java)|100%|:green_apple:|
|[UserService.java](https://github.com/itmo-events-app/event-app/blob/efe64da851a49447f2fa8e637e2311fb2b41fa7b/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FUserService.java)|90.63%|:green_apple:|
|[RoleService.java](https://github.com/itmo-events-app/event-app/blob/efe64da851a49447f2fa8e637e2311fb2b41fa7b/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FRoleService.java)|71.93% **`-17.98%`**|:green_apple:|
|[EventRoleService.java](https://github.com/itmo-events-app/event-app/blob/efe64da851a49447f2fa8e637e2311fb2b41fa7b/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FEventRoleService.java)|23.49% **`-46.64%`**|:x:|
|[RoleController.java](https://github.com/itmo-events-app/event-app/blob/efe64da851a49447f2fa8e637e2311fb2b41fa7b/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fcontroller%2FRoleController.java)|6.35% **`-38.62%`**|:x:|

**Comment by IP-13** [comment]
|Overall Project|77.42% **`-5.91%`**|:green_apple:|
|:-|:-|:-:|
|Files changed|36.38%|:x:|
<br>

|File|Coverage||
|:-|:-|:-:|
|[PrivilegeService.java](https://github.com/itmo-events-app/event-app/blob/1ebf1de2759395dc239b71deba7164c0f2497458/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FPrivilegeService.java)|100%|:green_apple:|
|[UserService.java](https://github.com/itmo-events-app/event-app/blob/1ebf1de2759395dc239b71deba7164c0f2497458/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FUserService.java)|90.63%|:green_apple:|
|[RoleService.java](https://github.com/itmo-events-app/event-app/blob/1ebf1de2759395dc239b71deba7164c0f2497458/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FRoleService.java)|71.93% **`-17.98%`**|:green_apple:|
|[EventRoleService.java](https://github.com/itmo-events-app/event-app/blob/1ebf1de2759395dc239b71deba7164c0f2497458/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FEventRoleService.java)|23.49% **`-46.64%`**|:x:|
|[RoleController.java](https://github.com/itmo-events-app/event-app/blob/1ebf1de2759395dc239b71deba7164c0f2497458/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fcontroller%2FRoleController.java)|6.35% **`-38.62%`**|:x:|

**Comment by IP-13** [comment]
|Overall Project|77.42% **`-5.91%`**|:green_apple:|
|:-|:-|:-:|
|Files changed|36.38%|:x:|
<br>

|File|Coverage||
|:-|:-|:-:|
|[PrivilegeService.java](https://github.com/itmo-events-app/event-app/blob/01843d7ad00deebd434d09d7f5a80f43827f9d7e/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FPrivilegeService.java)|100%|:green_apple:|
|[UserService.java](https://github.com/itmo-events-app/event-app/blob/01843d7ad00deebd434d09d7f5a80f43827f9d7e/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FUserService.java)|90.63%|:green_apple:|
|[RoleService.java](https://github.com/itmo-events-app/event-app/blob/01843d7ad00deebd434d09d7f5a80f43827f9d7e/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FRoleService.java)|71.93% **`-17.98%`**|:green_apple:|
|[EventRoleService.java](https://github.com/itmo-events-app/event-app/blob/01843d7ad00deebd434d09d7f5a80f43827f9d7e/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FEventRoleService.java)|23.49% **`-46.64%`**|:x:|
|[RoleController.java](https://github.com/itmo-events-app/event-app/blob/01843d7ad00deebd434d09d7f5a80f43827f9d7e/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fcontroller%2FRoleController.java)|6.35% **`-38.62%`**|:x:|

File: src/main/java/org/itmo/eventapp/main/repository/PrivilegeRepository.java [hunk 4]
```diff
@@ -13,4 +14,6 @@ public interface PrivilegeRepository extends JpaRepository<Privilege, Integer> {
 
     List<Privilege> findAllByType(PrivilegeType type);
 
+    Privilege findByName(PrivilegeName name);
```
[the remark this answers is not in the document]
Comment (IP-13) [reply src/main/java/org/itmo/eventapp/main/repository/PrivilegeRepository.java:17]: Есть такое, иначе все сломается. В отдельном ПР сделать нужно, этот уже так смерджим

File: src/main/java/org/itmo/eventapp/main/repository/UserRepository.java [hunk 5]
```diff
@@ -5,12 +5,11 @@
...
-import java.util.List;
 import java.util.Optional;
 
 @Repository
 public interface UserRepository extends JpaRepository<User, Integer> {
-    List<User> findAllByRole(Role role);
-
     Optional<User> findByUserLoginInfo_Login(String login);
+
+    boolean existsByRole(Role role);
```
Comment (IP-13) [inline src/main/java/org/itmo/eventapp/main/repository/UserRepository.java:14]: Лучше сделать existsBy...(имя параметра, а не по entity поиск делать).  В следующем ПР-е тоже сделать

File: src/main/java/org/itmo/eventapp/main/repository/EventRoleRepository.java [hunk 6]
```diff
@@ -13,7 +13,7 @@
 @Repository
 public interface EventRoleRepository extends JpaRepository<EventRole, Integer> {
 
-    List<EventRole> findAllByRole(Role role);
+    boolean existsByRole(Role role);
```
Comment (IP-13) [inline src/main/java/org/itmo/eventapp/main/repository/EventRoleRepository.java:16]: Лучше сделать existsBy...(имя параметра, а не по entity поиск делать).  В следующем ПР-е тоже сделать

File: src/main/java/org/itmo/eventapp/main/service/EventRoleService.java [hunk 7]
```diff
@@ -22,86 +24,83 @@ public class EventRoleService {
     private final EventRepository eventRepository;
     private final UserService userService;
     private final RoleService roleService;
-    private final List<String> notAssignableRoles = Arrays.asList("Администратор", "Читатель", "Организатор");
+    private final List<String> defaultOrganizationalRoles = Arrays.asList("Помощник", "Организатор");
```
Comment (IP-13) [inline src/main/java/org/itmo/eventapp/main/service/EventRoleService.java:27]: Лучше сделать константы, дальше все равно берется строкове представление константы.
notAssignableRoles.contains(role.getName())

File: src/main/java/org/itmo/eventapp/main/service/PrivilegeService.java [hunk 8]
```diff
@@ -20,11 +19,23 @@ public class PrivilegeService {
...
 
     public List<Privilege> getPrivilegeByType(PrivilegeType type) {
-        return privilegeRepository.findAllByType(type);
+        List<Privilege> privileges = privilegeRepository.findAllByType(type);
+
+        if (type.equals(PrivilegeType.SYSTEM)) {
+            Privilege assignOrganizerPrivilege =
+                    privilegeRepository.findByName(
+                            PrivilegeName.ASSIGN_ORGANIZER_ROLE);
+            if (null != assignOrganizerPrivilege)
```
Comment (IP-13) [inline src/main/java/org/itmo/eventapp/main/service/PrivilegeService.java:34]: Может поменять местами null, а то выглядит неудобно

File: src/main/java/org/itmo/eventapp/main/service/PrivilegeService.java [hunk 9]
```diff
@@ -20,11 +19,23 @@ public class PrivilegeService {
...
     }
 
     public List<Privilege> getPrivilegeByType(PrivilegeType type) {
-        return privilegeRepository.findAllByType(type);
+        List<Privilege> privileges = privilegeRepository.findAllByType(type);
+
+        if (type.equals(PrivilegeType.SYSTEM)) {
+            Privilege assignOrganizerPrivilege =
+                    privilegeRepository.findByName(
+                            PrivilegeName.ASSIGN_ORGANIZER_ROLE);
```
Comment (IP-13) [inline src/main/java/org/itmo/eventapp/main/service/PrivilegeService.java:33]: У этой привилегии нет типа? Или почему тут проверка, я пока не очень понимаю

File: src/test/java/org/itmo/eventApp/main/service/PrivilegeServiceTest.java [hunk 10]
```diff
@@ -0,0 +1,190 @@
...
+
+    @Test
+    @DisplayName("[getAll]-(Pos) Getting all privileges")
+    void getAllPrivilegesTest() {
+        // ====---- Setting up precondition ----====
+        cleanRolesPrivileges();
+        insertBasicFilling();
+
+        // ==========----  Execution  ----==========
+        List<Privilege> privileges = privilegeService.getAll();
```
Comment (IP-13) [inline src/test/java/org/itmo/eventApp/main/service/PrivilegeServiceTest.java:54]: Ну это же не интеграционные тесты, а модульные. Интеграционные тесты проверяют приложение от контроллера до БД, запросы отправляются через mockMvc

**Review by IP-13 — APPROVED** [review]

File: src/main/java/org/itmo/eventapp/main/service/PrivilegeService.java (hunk 9 above)
Comment (rimnvd) [reply src/main/java/org/itmo/eventapp/main/service/PrivilegeService.java:33]: Она и системная, и организационная, организатора мероприятия могут назначать и админ, и орг, поэтому она должна в обоих случаях выводиться при создании/редактировании роли

**Pull request merged**